### PR TITLE
fix: service toggle issue after switch focus

### PR DESCRIPTION
**Pull request by phlpsong**
close https://github.com/tisfeng/Easydict/issues/423

### CODE REVIEW

**Comment by tisfeng** [comment]
The code works, but it will cause this bug again https://github.com/tisfeng/Easydict/pull/326#issuecomment-1893017637

File: Easydict/NewApp/View/SettingView/Tabs/ServiceTab.swift [hunk 1]
```diff
@@ -121,28 +121,29 @@ private struct ServiceItems: View {
...
 private struct ServiceItemView: View {
     let service: QueryService
 
     @EnvironmentObject private var viewModel: ServiceTabViewModel
 
-    private var enabled: Binding<Bool> {
-        .init {
-            service.enabled
-        } set: { newValue in
-            guard service.enabled != newValue else { return }
```
Comment (tisfeng) [inline Easydict/NewApp/View/SettingView/Tabs/ServiceTab.swift:134]: This code is needed, we shouldn't post update notification if enable has no new value.

**Review by tisfeng — CHANGES_REQUESTED** [review]

File: Easydict/NewApp/View/SettingView/Tabs/ServiceTab.swift (hunk 1 above)
Comment (phlpsong) [reply Easydict/NewApp/View/SettingView/Tabs/ServiceTab.swift:134]: The logic moved inside of `onChangeValue` modifier.

**Comment by phlpsong** [comment]
> The code works, but it will cause this bug again [#326 (comment)](https://github.com/tisfeng/Easydict/pull/326#issuecomment-1893017637)

fixed.

**Comment by tisfeng** [comment]
ok, looks good.

**Review by tisfeng — APPROVED** [review]

**Review by Jerry23011 — APPROVED** [review]
LGTM. Also fixed a bug where the toggle state is not updating after switching windows.

**Review by AkaShark — APPROVED** [review]

**Pull request merged**